File: dash_docs/chapters/basic_callbacks/examples/simple-callback.R
library(dash)
library(dashCoreComponents)
library(dashHtmlComponents)

app <- Dash$new()

app$layout(
  htmlDiv(
    list(
      dccInput(id='my-id', value='initial value', type='text'),
      htmlDiv(id='my-div')
    )
  )
)

app$callback(
  output=list(id='my-div', property='children'),
  params=list(input(id='my-id', property='value')),
  function(input_value) {
    sprintf("You've entered \"%s\"", input_value)
  })

app$run_server()

File: dash_docs/chapters/basic_callbacks/examples/simple-callback.jl
using Dash, DashHtmlComponents, DashCoreComponents

app = dash()

app.layout = html_div() do
    dcc_input(id = "input-3", value = "initial value", type = "text"),
    html_div(id="output")
end

callback!(app, Output("output", "children"), Input("input-3", "value")) do input_value
    "You've entered $(input_value)"
end

run_server(app, "0.0.0.0", debug=true)
